src/loop/typer_intf.ml

```diff
@@ -143,17 +143,26 @@ module type Typer_Full = sig
   (** This signature includes the requirements to instantiate the {Pipes.Make:
       functor*)
 
+  val init :
+    ?ty_state:ty_state ->
+    ?smtlib2_forced_logic:string option ->
+    ?additional_builtins:(lang -> builtin_symbols) ->
+    state -> state
+
+  val additional_builtins : (lang -> builtin_symbols) key
```

Comment:
My first draft actually passed `state` instead of `lang` here, but I was not sure about you wanting to allow the builtins to depend on the file being parsed, it feels hackish.

I am fine with adding `state` there. Do you prefer the type be `state -> lang -> builtin_symbols` or just `state -> builtin_symbols`?

Reply:
Let's give both the `state` and the `lang`.

Note that users can already insert code in a pipeline that would match on anything available in the state, and change things (either the `additional_builtins` key, or ref) depending on things that can be more or less reasonable. For instance, users could implement some kind of extensions systems for a language where files would have additional conventions (e.g. `some_file_name.extensions.smt2`) to indicate to add some typing extensions, or some users might change some behavior depending on whether we will later verify a model corresponding to the file (which is actually something that dolmen already does). Anyway, we cannot prevent users from shooting themselves in the foot, and there are legitimate reasons to lookup things into the state in order to decide how to type some file (plus it can already be done currently, it's just a bit harder), ^^

Reply:
Oh I know, I may use it in Alt-Ergo to restrict bulitins to the builtin preludes. And yeah, good point about being able to do it from the pipeline, let's pass in the `state`.